### DISCUSSION TITLE
Candle width doesn't adjust well upon zoom

### DISCUSSION THREAD

**Discussion by somguyth** [reactivemarkets/react-financial-charts#534] [General]
Zooming out a little, and the candles look like [this](https://ibb.co/WvdV8Y0). If I zoom in very slightly (one click of the mouse wheel), it'll then look like [this](https://ibb.co/LNhXx2Z).

The second chart looks good, but the first one is not good and it detracts from the UX. The candle body is simply too small and there's too much whitespace between the candles.

If we compare it to TradingView, all the candles there are really nicely packed together irrespective of the zoom level. I just opened TradingView now to compare, and it does a good job at eliminating all whitespace. [This is zoomed out TradingView](https://ibb.co/LY9VP4q) for comparison.

Is there a way we can fix it so that the candle body adjusts better to the level of zoom, by reducing the amount of whitespace between the candles?

**Comment by markmcdowell** [comment]
There's 2 properties on the `CandlestickSeries` that describes this behavior: `widthRatio` and `width`. `widthRatio` defaults to 0.8 which is why you get small gaps between candles, you could change this to see if it works for you. Otherwise you can supply a custom width function to calculate the width of each candle.
